### DISCUSSION TITLE
dynamically creating many w2field combos in grid toolbar

### DISCUSSION THREAD

**Discussion by jankrnavek** [vitmalina/w2ui#2485] [General]
Hello all,
In my app I dynamically create/remove many  w2field combo objects as needed (according existing colum fields).

I do it in two steps.
At first  I create toolbar items  over `w2ui.grid.toolbar.add(itemDefinition)`
then I initialize w2field object something like
```
const inputEl = query(parentToolbarEl).find('input').get(0)

new w2field('combo', {
			el: inputEl,
			items: getItemsFuction,
			filter: false,
			})
```
It all seeems to work but...

I have three questions regarding   w2field combos.

1) Is that approach ok?
2) How to properly remove all w2field objects (toolbar items)  to avoid memory problems? Is    `w2ui.grid.toolbar.remove(itemID)` enough considering w2field combo object?
3)  Can I somehow programmatically refresh w2field object because of the fresh items in dropdown item list?  
There is maybe some issue. 
 If I focuse input field (combo) , sometimes the dropdown item list don´t show all items getting from  `items: getItemsFuction` .  Re-focus e.g. going to next the input field and back  to origin input  helps.

Thank in advance.
H.

**Comment by vitmalina** [comment]
1. Yes, you can re-init drop down fields, but I think the best way would be like is is shown here: http://localhost:3500/demos/#/toolbar/14, see how date field is initialized. I think you can init drop down in the similar way.

2. Element itself has link to the `w2field` object, removing element from the dom should mark it for cargage collection and it should be deleted

3. As for items in drop down. I think you can just update items without re-initializing the control. If you apply w2field to an input then you can do this
```js
// update items directly and it should work
$(input).get(0)._w2field.options.items[0].text = '11';
```
